v0/destinations/hs/transform.js

```diff
@@ -1,351 +1,69 @@
+      } else {
+        response.push(
+          await processLegacyIdentify(message, destination, propertyMap)
         );
       }
-      throw new CustomError(
-        "Failed to get hubspot properties : indvalid response",
-        500
-      );
-    }
-
-    const propertyMap = {};
-    response.data.forEach(element => {
-      propertyMap[element.name] = element.type;
-    });
-    hubSpotPropertyMap = propertyMap;
-  }
-  return hubSpotPropertyMap;
-}
 
-async function getTransformedJSON(
-  message,
-  mappingJson,
-  destination,
-  propertyMap
-) {
-  const rawPayload = {};
-  const sourceKeys = Object.keys(mappingJson);
-  const traits = getTraits(message);
-
-  if (traits) {
-    const traitsKeys = Object.keys(traits);
-    if (!propertyMap) {
-      propertyMap = await getProperties(destination);
+      // handle CRM Association v3
+      // const traits = getFieldValueFromMessage(message, "traits");
+      // if (traits.hubspot) {
+      //   response.push(
+      //     await processCRMCustomObjects(message, destination, traits)
+      //   );
+      // }
+      break;
     }
-    sourceKeys.forEach(sourceKey => {
-      if (get(traits, sourceKey)) {
-        set(rawPayload, mappingJson[sourceKey], get(traits, sourceKey));
-      }
-    });
-    traitsKeys.forEach(traitsKey => {
-      const hsSupportedKey = getKey(traitsKey);
-      if (!rawPayload[traitsKey] && propertyMap[hsSupportedKey]) {
-        let propValue = traits[traitsKey];
-        if (propertyMap[hsSupportedKey] === "date") {
-          const time = propValue;
-          const date = new Date(time);
-          date.setUTCHours(0, 0, 0, 0);
-          propValue = date.getTime();
-        }
-        rawPayload[hsSupportedKey] = propValue;
-      }
-    });
-  }
-  return { ...rawPayload };
-}
-
-function getPropertyValueForIdentify(propMap) {
-  return Object.keys(propMap).map(key => {
-    return { property: key, value: propMap[key] };
-  });
-}
-
-function responseBuilderSimple(payload, message, eventType, destination) {
-  let endpoint = "https://track.hubspot.com/v1/event";
-  let params = {};
-
-  const response = defaultRequestConfig();
-  response.method = defaultGetRequestConfig.requestMethod;
-
-  if (eventType !== EventType.TRACK) {
-    const traits = getFieldValueFromMessage(message, "traits");
-    const { email } = traits;
-    const { apiKey } = destination.Config;
-    params = { hapikey: apiKey };
-    if (get(message, MappedToDestinationKey)) {
-      const { objectType } = getDestinationExternalIDInfoForRetl(message, "HS");
-      endpoint = `https://api.hubspot.com/crm/v3/objects/${objectType}`;
-      response.method = defaultPostRequestConfig.requestMethod;
-      response.body.JSON = removeUndefinedValues({ properties: traits });
-    } else {
-      if (email) {
-        endpoint = `https://api.hubapi.com/contacts/v1/contact/createOrUpdate/email/${email}`;
+    case EventType.TRACK:
+      if (destination.Config.apiVersion === "newApi") {
```

Comment:
Lets not use this nomenclature for version management use hubspot standard v0/v1/v2 .. what ever we are using

Reply:
Done